Spravo/AppDelegate.swift

```diff
@@ -45,4 +50,25 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func applicationWillTerminate(_ application: UIApplication) {
     }
+    
+    @objc func checkForReachability(notification: Notification) {
+        guard let reachability = notification.object as? Reachability,
+            let rootVC = self.window?.rootViewController else { return }
+        if !reachability.isReachable() {
+            updateUIonMainThread {
+                guard let errorController = Storyboard.service.controller(withClass: ErrorScreenVC.self) else { return }
+                errorController.image = UIImage(named: "Disconnected")
+                errorController.text = NSLocalizedString("ImportPhoneContacts.ErrorInternetConnection", comment: "Request to check internet connection")
+                rootVC.modalPresentationStyle = .overCurrentContext
+                rootVC.present(errorController, animated: true, completion: nil)
+            }
+        } else {
+            if rootVC is UINavigationController {
+                let topController = (rootVC as! UINavigationController).visibleViewController
+                if topController is ErrorScreenVC {
+                    topController?.dismiss(animated: true, completion: nil)
```

Comment:
error page is presented on Main thread, but you are dismissing it not on Main thread. why?

Reply:
Fix: controller dismiss on main thread